addons/purchase_edi_ubl_bis3/models/purchase_edi_common.py

```diff
@@ -0,0 +1,184 @@
+}
+
+# -------------------------------------------------------------------------
+# ELECTRONIC ADDRESS SCHEME (EAS), see https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/
+# -------------------------------------------------------------------------
+EAS_MAPPING = {
+    'AD': {'9922': 'vat'},
+    'AL': {'9923': 'vat'},
+    'AT': {'9915': 'vat'},
+    'AU': {'0151': 'vat'},
+    'BA': {'9924': 'vat'},
+    'BE': {'0208': 'company_registry'},
+    'BG': {'9926': 'vat'},
+    'CH': {'9927': 'vat'},
+    'CY': {'9928': 'vat'},
+    'CZ': {'9929': 'vat'},
+    'DE': {'9930': 'vat'},
+    'DK': {'0184': 'company_registry', '0198': 'vat'},
+    'EE': {'9931': 'vat'},
+    'ES': {'9920': 'vat'},
+    'FI': {'0213': 'vat'},
+    'FR': {'0009': 'siret', '9957': 'vat'},
+    'SG': {'0195': 'l10n_sg_unique_entity_number'},
+    'GB': {'9932': 'vat'},
+    'GR': {'9933': 'vat'},
+    'HR': {'9934': 'vat'},
+    'HU': {'9910': 'vat'},
+    'IE': {'9935': 'vat'},
+    'IS': {'0196': 'vat'},
+    'IT': {'0211': 'vat', '0210': 'l10n_it_codice_fiscale'},
+    'JP': {'0221': 'vat'},
+    'LI': {'9936': 'vat'},
+    'LT': {'9937': 'vat'},
+    'LU': {'9938': 'vat'},
+    'LV': {'9939': 'vat'},
+    'MC': {'9940': 'vat'},
+    'ME': {'9941': 'vat'},
+    'MK': {'9942': 'vat'},
+    'MT': {'9943': 'vat'},
+    # Do not add the vat for NL, since: "[NL-R-003] For suppliers in the Netherlands, the legal entity identifier
+    # MUST be either a KVK or OIN number (schemeID 0106 or 0190)" in the Bis 3 rules (in PartyLegalEntity/CompanyID).
+    'NL': {'0106': None, '0190': None},
+    'NO': {'0192': 'l10n_no_bronnoysund_number'},
+    'NZ': {'0088': 'company_registry'},
+    'PL': {'9945': 'vat'},
+    'PT': {'9946': 'vat'},
+    'RO': {'9947': 'vat'},
+    'RS': {'9948': 'vat'},
+    'SE': {'0007': 'vat'},
+    'SI': {'9949': 'vat'},
+    'SK': {'9950': 'vat'},
+    'SM': {'9951': 'vat'},
+    'TR': {'9952': 'vat'},
+    'VA': {'9953': 'vat'},
+}
+
+
+class PurchaseEdiCommon(models.AbstractModel):
+    _name = "purchase.edi.common"
+    _description = "Common functions for EDI documents"
+
+    # -------------------------------------------------------------------------
+    # HELPERS
+    # -------------------------------------------------------------------------
+
+    def format_float(self, amount, precision_digits):
+        if amount is None:
+            return None
+        return float_repr(float_round(amount, precision_digits), precision_digits)
+
+    def _get_currency_decimal_places(self, currency_id):
+        # Allows other documents to easily override in case there is a flat max precision number
+        return currency_id.decimal_places
+
+    def _get_uom_unece_code(self, line):
+        """
+        list of codes: https://docs.peppol.eu/poacc/billing/3.0/codelist/UNECERec20/
+        or https://unece.org/fileadmin/DAM/cefact/recommendations/bkup_htm/add2c.htm (sorted by letter)
+        """
+        xmlid = line.product_uom.get_external_id()
```

Comment:
What if the UoM was created by the user and don't have xml_id? Maybe we could do an extra check on name

Reply:
returning just unit seems a bit dangerous